### PR TITLE
AZP/RELEASE: Rework JUCX verification & release processes

### PR DESCRIPTION
**Pull request by Alexey-Rivkin**
## What
Summary of actions per trigger:

| Trigger                          | Actions                                        |
| -------------------------------- | ---------------------------------------------- |
| **PR request**                   | 1. Ensure that the JAR compiles.                |
|                                  | 2. Publish to local FS.                         |
|                                  | 3. Run benchmark tests using the locally published JAR. |
| **Merge to master or release branches** | Publish Snapshot to Maven Central repo (mutable). |
| **Release (including RCs)**      |  Publish release JAR to Maven Central repo (immutable). |

## Why
The purpose is to detect possible JUCX compilation and publishing issues on the PR level.

## How:
Publish to Maven Central is defined at `jucx-build.yml`:

- The pipeline "UCX release" calls it with the target "publish-release".
- The pipeline "UCX snapshot" calls it with the target "publish-snapshot".

The local publish step is at `jucx-test.yml`, which is only called by the "UCX PR" pipeline.

### Additional fixes:
1. Migtate from CentOS-7 to CentOS-8 due to obsolete GPG agent.
2. Use 'pinentry' to GPG sign JAR files.
4. Add a filename suffix for non-x86 platforms.
5. Minor refactoring to simplify the code and improve readability.

### CODE REVIEW

**Comment by Alexey-Rivkin** [comment]
### Verification:
[UCX snapshot build](https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=75778)
[UCX release build](https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=75791)
[UCX PR build](https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=75792)

File: bindings/java/src/main/native/Makefile.am [hunk 1]
```diff
@@ -50,6 +50,10 @@ $(JUCX_GENERATED_H_FILES): $(STAMP_FILE)
 
 # Return CPU architecure in Java format
 arch = $(if $(filter x86_64,$(shell uname -p)),amd64,$(shell uname -p))
+
+# Append non-x86_64 arch as filename suffix
+suffix = $(if $(filter-out x86_64,$(shell uname -p)),-$(shell uname -p))
```
Comment (yosefe) [inline bindings/java/src/main/native/Makefile.am:55]: don't we have multiple archs in same jar?

File: buildlib/azure-pipelines.yml [hunk 2]
```diff
@@ -36,25 +34,21 @@ stages:
...
           demands: ucx_docker
+          target: publish-snapshot
 
       - template: jucx/jucx-build.yml
         parameters:
           arch: aarch64
           container: centos8_cuda11_aarch64
           demands: ucx-arm64
-          
-      - template: jucx/jucx-publish.yml
```
Comment (yosefe) [inline buildlib/azure-pipelines.yml:55]: do we still use the file jucx-publish.yml? can we remove it?

File: buildlib/jucx/jucx-test.yml [hunk 3]
```diff
@@ -20,13 +21,11 @@ jobs:
 
     steps:
       - checkout: self
-        fetchDepth: 100
         clean: true
         displayName: Checkout
       - bash: |
           set -xeE
           source buildlib/az-helpers.sh
-          check_gpu ${{ parameters.name }}
```
Comment (yosefe) [inline buildlib/jucx/jucx-test.yml:29]: can we still check that gpu is present when we use ucx_gpu machine?

File: bindings/java/src/main/native/Makefile.am (hunk 1 above)
Comment (Alexey-Rivkin) [reply bindings/java/src/main/native/Makefile.am:55]: ARM JARs have the specific platform name appended to their filenames, while x86_64 JARs retain their original filenames. This decision was made after a discussion in an email thread about separating JARs by platform.

File: buildlib/azure-pipelines.yml (hunk 2 above)
Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines.yml:56]: jucx-publish.yml is still used for Maven Central publishing.

File: buildlib/jucx/jucx-test.yml (hunk 3 above)
Comment (Alexey-Rivkin) [reply buildlib/jucx/jucx-test.yml]: Of course. Merged from master and got it back.

File: buildlib/jucx/jucx-build.yml [hunk 4]
```diff
@@ -22,6 +22,8 @@ jobs:
     steps:
       - checkout: self
         clean: true
+        # fetchDepth: 100
```
Comment (yosefe) [inline buildlib/jucx/jucx-build.yml:25]: remove this line?

File: buildlib/jucx/jucx-test.yml [hunk 5]
```diff
@@ -21,12 +22,14 @@ jobs:
 
     steps:
       - checkout: self
+        # fetchDepth: 100
```
Comment (yosefe) [inline buildlib/jucx/jucx-test.yml:25]: remove this line?

**Review by yosefe — COMMENTED** [review]
LGTM, can we uncomment the fecthDepth?

**Comment by yosefe** [comment]
@Alexey-Rivkin maybe also need to merge with latest master branch?

**Comment by Alexey-Rivkin** [comment]
Added `git fetch --unshallow` to resolve the error "No tags can describe \<SHA>".

**Review by yosefe — APPROVED** [review]

**Pull request merged**